### PR TITLE
Add Microsoft.M365SecurityAndCompliance RP

### PR DESCRIPTION
**Pull request by sonalverm**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [Y] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ Y] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sonalverm** [comment]
@raychen ,@jianyxi can you please help me fix the errors related to SDK azure resource manager schemas?

**Comment by pilor** [comment]
Please describe the delta between what was approved in the private repo (https://github.com/Azure/azure-rest-api-specs/pull/12812) and what is in this PR if any.

**Comment by sonalverm** [comment]
This is a rename PR request from Microsoft.SecurityAndCompliance to Microsoft.M365SecurityAndCompliance as suggested by @ravbhatnagar. 
The changes are already merged in master branch of public repo.

**Comment by sonalverm** [comment]
@jianyexi can you please help me resolve these errors?

**Comment by jianyexi** [comment]
@leni-msft  @ArcturusZhang  could you look at the CI failure ?

**Comment by pilor** [comment]
Approving from ARM side as author indicated this is purely a rename of the RP from the existing approved content from #12812

File: specification/m365securityandcompliance/resource-manager/readme.go.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+output-folder: $(go-sdk-folder)/services/$(namespace)/mgmt/2021-03-25-preview/$(namespace)
+```
+
+### Tag: package-2021-03-25-preview and go
+
+These settings apply only when `--tag=package-2021-03-25-preview --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag)=='package-2021-03-25-preview' && $(go)
+output-folder: $(go-sdk-folder)/services/$(namespace)/mgmt/2021-03-25-preview/$(namespace)
```
Comment (ArcturusZhang) [inline specification/m365securityandcompliance/resource-manager/readme.go.md:35]:
```suggestion
output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2021-03-25-preview/$(namespace)
```

File: specification/m365securityandcompliance/resource-manager/readme.go.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  - tag: package-2021-03-25-preview
+```
+
+### Tag: package-2021-03-25-preview and go
+
+These settings apply only when `--tag=package-2021-03-25-preview --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag)=='package-2021-03-25-preview' && $(go)
+output-folder: $(go-sdk-folder)/services/$(namespace)/mgmt/2021-03-25-preview/$(namespace)
```
Comment (ArcturusZhang) [inline specification/m365securityandcompliance/resource-manager/readme.go.md:26]: Preview packages must be generated to the preview subdirectory, as said in the CI error message
```suggestion
output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2021-03-25-preview/$(namespace)
```

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Comment by sonalverm** [comment]
@jianyexi could you please merge this PR?

**Comment by jianyexi** [comment]
@ArcturusZhang could you approve it ?

File: specification/m365securityandcompliance/resource-manager/readme.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+``` yaml $(tag) == 'package-2021-03-25-preview'
+input-file:
+- Microsoft.M365SecurityAndCompliance/preview/2021-03-25-preview/common-types.json
+- Microsoft.M365SecurityAndCompliance/preview/2021-03-25-preview/privateLinkServicesForEDMUpload.json
+- Microsoft.M365SecurityAndCompliance/preview/2021-03-25-preview/privateLinkServicesForM365ComplianceCenter.json
+- Microsoft.M365SecurityAndCompliance/preview/2021-03-25-preview/privateLinkServicesForM365SecurityCenter.json
+- Microsoft.M365SecurityAndCompliance/preview/2021-03-25-preview/privateLinkServicesForO365ManagementActivityAPI.json
+- Microsoft.M365SecurityAndCompliance/preview/2021-03-25-preview/privateLinkServicesForSCCPowershell.json
+- Microsoft.M365SecurityAndCompliance/preview/2021-03-25-preview/privateLinkServicesForMIPPolicySync.json
+```
```
Comment (ArcturusZhang) [inline specification/m365securityandcompliance/resource-manager/readme.md:61]: Why we are having two different tags with the same name here?
Could you please double confirm this? @sonalverm

File: specification/m365securityandcompliance/resource-manager/readme.go.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+### Tag: package-2021-03-25-preview and go
+
+These settings apply only when `--tag=package-2021-03-25-preview --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag)=='package-2021-03-25-preview' && $(go)
+output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2021-03-25-preview/$(namespace)
+```
```
Comment (ArcturusZhang) [inline specification/m365securityandcompliance/resource-manager/readme.go.md:36]: We need to remove one of the duplicated tags here

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/m365securityandcompliance/resource-manager/readme.md (hunk 3 above)
Comment (sonalverm) [reply specification/m365securityandcompliance/resource-manager/readme.md:61]: I'll remove one of them.

File: specification/m365securityandcompliance/resource-manager/readme.go.md (hunk 4 above)
Comment (sonalverm) [reply specification/m365securityandcompliance/resource-manager/readme.go.md:36]: I'll remove one of them.

**Comment by sonalverm** [comment]
@ArcturusZhang could you please approve it?

**Comment by jianyexi** [comment]
@sonalverm  seems you should add your new swagger 'preview/2021-03-25-preview/examples/MIPPolicySyncServiceList.json' to  readme tag to fix the avocado error

**Comment by sonalverm** [comment]
@ArcturusZhang  @jianyexi  could you please approve it?

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**